sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -1946,7 +1952,15 @@ class CompactBlockProcessor internal constructor(
                 }
                 when (val response = downloader.fetchTransaction(rawTransactionId)) {
                     is Response.Success -> {
-                        transactionDataResult = response.result.data
+                        val currentMinedHeight = when (response.result) {
+                            is RawTransactionUnsafe.MainChain -> runCatching {
```

Comment:
@str4d Do we need this check? The only way how this fails is due to the height below the sapling activation height. Do we expect such heights here?

Reply:
Eventually we'll want to support heights below Sapling activation. But this specific catch was kept from my earlier iteration of this PR, before you pointed out that very similar checks were present in `BlockHeightUnsafe`. Now that the height field is correctly handled, we maybe don't need this, but it shouldn't hurt.